Mathlib/RingTheory/MvPolynomial/MonomialOrder.lean

```diff
@@ -71,16 +71,42 @@ variable {R : Type*} [CommSemiring R]
 
 variable (m) in
 /-- the degree of a multivariate polynomial with respect to a monomial ordering -/
-def degree {R : Type*} [CommSemiring R] (f : MvPolynomial σ R) : σ →₀ ℕ :=
+def degree (f : MvPolynomial σ R) : σ →₀ ℕ :=
   m.toSyn.symm (f.support.sup m.toSyn)
 
 variable (m) in
 /-- the leading coefficient of a multivariate polynomial with respect to a monomial ordering -/
-def leadingCoeff {R : Type*} [CommSemiring R] (f : MvPolynomial σ R) : R :=
+def leadingCoeff (f : MvPolynomial σ R) : R :=
   f.coeff (m.degree f)
 
 @[deprecated (since := "2025-01-31")] alias lCoeff := leadingCoeff
 
+theorem leadingCoeff_eq {f : MvPolynomial σ R} : m.leadingCoeff f = f.coeff (m.degree f) := rfl
```

Comment:
I think we don't really want that lemma?

Reply:
Oddly enough, I needed it because `rw [leadingCoeff]` or `simp only [leadingCoeff]` had not worked.

Reply:
entering `rw [MonomialOrder.leadingCoeff]` or `rw [MonomialOrder.Monic]` suffices.